admin/src/app/Components/images-for-admin/images-for-admin.component.ts

```diff
@@ -1,44 +1,119 @@
 import { Component, Input, OnInit } from '@angular/core';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { HttpErrorResponse } from '@angular/common/http';
+import { Observable } from 'rxjs';
 
 import { ImageService } from 'src/app/image.service';
 import { Image } from 'src/app/Dtos/image';
+import { HotelService } from 'src/app/hotel.service';
+import { RoomService } from '../../room.service';
+import { Room } from 'src/app/Dtos/room';
+import { Hotel } from 'src/app/Dtos/hotel';
 
 @Component({
   selector: 'app-images-for-admin',
   templateUrl: './images-for-admin.component.html',
   styleUrls: ['./images-for-admin.component.css']
 })
 export class ImagesForAdminComponent implements OnInit {
+  @Input() public hotel?: Hotel;
+  @Input() public room?: Room;
+
+  public hotelId?: number;
+  public roomId?: number;
+
   public progress: number = 0;
   public images: Image[] = [];
-  @Input() public hotelId?: number;
 
   public constructor(
-    private readonly imageService: ImageService
-  ) { 
+    private readonly imageService: ImageService,
+    private readonly snackBar: MatSnackBar,
+    private readonly hotelService: HotelService,
+    private readonly roomService: RoomService
+  ) {
   }
 
   public ngOnInit(): void {
+    this.hotelId = this.hotel?.id;
+    this.roomId = this.room?.id;
     this.fetchImages();
   }
   
   public createImagePath(image: Image): string {
-    if (this.hotelId === undefined) {
+    if (this.hotel === undefined) {
       return '';
     }
-    return this.imageService.createImagePath(this.hotelId, image);
+    return this.imageService
+      .createImagePath(
+        this.hotel.id, 
+        image.id,
+        this.room?.id
+      );
   }
 
   public addImage(imageId: number): void {
     this.images.push({ id: imageId } as Image);
   }
 
+  public changeMainImage(image: Image): void {
+    if (this.hotel === undefined) {
+      return;
+    }
+    let changeMain$: Observable<string>;
+    if (this.room) {
+      this.room.mainImageId = image.id;
+      changeMain$ = this.roomService.editRoom(this.hotel.id, this.room.id, this.room);
+    } else {
+      this.hotel.mainImageId = image.id;
+      changeMain$ = this.hotelService.editHotel(this.hotel.id, this.hotel);
+    }
+    changeMain$      
+      .subscribe(
+        () => {
+          this.openSnackBar('Main image was successfully changed');
+        },
+        (serverError: HttpErrorResponse) => {
+          this.openSnackBar(serverError.error as string);
+        }
+      );
+  }
+
+  public deleteImage(image: Image): void {
+    if (this.hotel === undefined) {
+      return;
+    }
+    this.imageService
+      .deleteImage(this.hotel.id, image.id, this.room?.id)
+      .subscribe(
+        () => {
+          if (this.room?.mainImageId === image.id) {
+            this.room.mainImageId = undefined;
+          }
+          this.images = this.images.filter(img => img !== image);
```

Comment:
I you suggest to use `id`s

```suggestion
          this.images = this.images.filter(img => img.id !== image.id);
```